render/gles2/meson.build

```diff
@@ -1,3 +1,12 @@
+glesv2 = dependency('glesv2', required : false)
```

Comment:
If the user has explicitely selected the GLES2 backend via the `renderers` option, then the `glesv2` dependency is required.

Reply:
but if only the auto is selected, we can't make it be required here. I could make it throw an error if the gles2 is selected

Reply:
This should do it:

```
glesv2 = dependency('glesv2', required: get_option('renderers').contains('gles2'))
```

aka. required only if the GLES2 renderer was explicitly enabled by the user.